### PR TITLE
Add wgsmetrics clean

### PR DESCRIPTION
**Pull request by Gwennid**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
_branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

## Description of changes

This PR adds an additional QC: computation of coverage statistics for the entire genome and for chromosome Y. This can be used to estimate the sex of the sample (note that calculating the ratio is not included in the PR, but the metrics files can be used to do that).
The coverage is computed with Sentieon [WgsMetricsAlgo](https://github.com/nf-core/raredisease/blob/dev/modules/local/sentieon/wgsmetricsalgo.nf) alternatively Picard's [CollectWgsMetrics](https://github.com/nf-core/modules/blob/master/modules/nf-core/picard/collectwgsmetrics/main.nf) and is integrated to the [qc_bam](https://github.com/nf-core/raredisease/blob/dev/subworkflows/nf-core/qc_bam.nf) subworkflow. Since Sentieon WgsMetricsAlgo is a local module, the qc_bam subworkflow was moved to local subworkflows.
The interval list files for Picard's CollectWgsMetrics are provided as assets.

Comment: the subworkflow was not tested for Sentieon yet.

### CODE REVIEW

**Review by sima-r — COMMENTED** [review]
I think we can remove line 54 and in line 55 simply use bam_bai instead of mosdepth_input_bams. Looks they're the same.

**Comment by Gwennid** [comment]
> I think in the file qc_bam, we can remove line 54 and in line 55 simply use bam_bai instead of mosdepth_input_bams. Looks they're the same.

@ramprasadn Do you validate @sima-r 's suggestion?

**Comment by ramprasadn** [comment]
@Gwennid Yupe, I agree with @sima 😄 
But it should not be an issue anymore as I had already fixed it my other PR (#277), and I just merged that PR into dev.

File: subworkflows/local/qc_bam.nf [hunk 1]
```diff
@@ -43,12 +50,34 @@ workflow QC_BAM {
...
+            ch_cov_y = PICARD_COLLECTWGSMETRICS_Y.out.metrics
+            ch_versions = ch_versions.mix(PICARD_COLLECTWGSMETRICS_Y.out.versions)
+        } else if ( aligner == "sentieon" ) {
+            SENTIEON_WGSMETRICS ( bam_bai, fasta, fai)
+            ch_cov = SENTIEON_WGSMETRICS.out.wgs_metrics
+            ch_versions = ch_versions.mix(SENTIEON_WGSMETRICS.out.versions)
+            SENTIEON_WGSMETRICS_Y ( bam_bai, fasta, fai)
+            ch_cov_y = SENTIEON_WGSMETRICS_Y.out.wgs_metrics
+            ch_versions = ch_versions.mix(SENTIEON_WGSMETRICS_Y.out.versions)
+        } else {
```
[the remark this answers is not in the document]
Comment (ramprasadn) [reply subworkflows/local/qc_bam.nf]: If the user doesn't provide one it will just assume the default to be bwamem2

**Review by sima-r — APPROVED** [review]
I have run it and it worked! So LGTM :)

**Pull request merged**